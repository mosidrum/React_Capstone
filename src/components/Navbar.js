import React, { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { AiOutlineMenu } from 'react-icons/ai';
import logo from '../assets/blue-earth-removebg-preview.png';
import '../css/navbar.css';

const Navbar = () => {
  const [isOpen, setIsOpen] = useState(false);
  const [isSmallScreen, setIsSmallScreen] = useState(false);

  useEffect(() => {
    const handleResize = () => {
      setIsSmallScreen(window.innerWidth <= 800);
    };

    handleResize();

    window.addEventListener('resize', handleResize);

    return () => {
      window.addEventListener('resize', handleResize);
    };
  }, []);

  const handleClick = () => {
    if (isSmallScreen) {
      setIsOpen(!isOpen);
    }
  };

  return (
    <header className={`${isOpen ? 'open' : ''}`}>
      <div className="left">
        <img src={logo} alt="planet" />
        <NavLink to="/"><h1>Global village</h1></NavLink>
      </div>

      <div
        className="menu-bar"
        onClick={handleClick}
        onKeyDown={handleClick}
        role="button"
        tabIndex={0}
      >
        <AiOutlineMenu />
      </div>

      <div className="right">
        <NavLink
          className="home"
          to="/"
          onClick={handleClick}
        >
          Home
        </NavLink>
        <NavLink
          className="search-nav"
          to="/search"
          onClick={handleClick}
        >
          Search
        </NavLink>
        <NavLink
          className="back"
          to="/"
          onClick={handleClick}
        >
          Back
        </NavLink>
      </div>
    </header>
  );
};

export default Navbar;
