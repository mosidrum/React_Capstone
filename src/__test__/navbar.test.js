import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  it('renders the Navbar component', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const logoElement = screen.getByAltText('planet');
    expect(logoElement).toBeInTheDocument();

    const homeLink = screen.getByText('Home');
    expect(homeLink).toBeInTheDocument();

    const searchLink = screen.getByText('Search');
    expect(searchLink).toBeInTheDocument();

    const backLink = screen.getByText('Back');
    expect(backLink).toBeInTheDocument();

    const tree = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
