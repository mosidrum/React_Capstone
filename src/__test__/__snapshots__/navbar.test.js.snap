// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the Navbar component 1`] = `
<header
  className=""
>
  <div
    className="left"
  >
    <img
      alt="planet"
      src="blue-earth-removebg-preview.png"
    />
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Global village
      </h1>
    </a>
  </div>
  <div
    className="menu-bar"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </div>
  <div
    className="right"
  >
    <a
      aria-current="page"
      className="home active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="search-nav"
      href="/search"
      onClick={[Function]}
    >
      Search
    </a>
    <a
      aria-current="page"
      className="back active"
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
  </div>
</header>
`;
